java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java

```diff
@@ -152,130 +130,380 @@ private void execute(String signature, SquidClassLoader classLoader) {
+      case IAND:
+      case LAND:
+      case IOR:
+      case LOR:
+      case IXOR:
+      case LXOR:
+        pop = programState.unstackValue(2);
+        Preconditions.checkState(pop.values.size() == 2, "Bitwise instruction needs 2 values on stack");
+        sv = constraintManager.createBinarySymbolicValue(instruction, pop.valuesAndSymbols);
+        programState = pop.state.stackValue(sv).addConstraint(sv, ObjectConstraint.NOT_NULL);
+        break;
+      case IINC:
+        int index = instruction.operand;
+        SymbolicValue existing = programState.getValue(index);
+        Preconditions.checkNotNull(existing, "Local variable " + index + " not found");
+        sv = constraintManager.createSymbolicValue(instruction);
+        programState = programState.put(index, sv).addConstraint(sv, ObjectConstraint.NOT_NULL);
+        break;
+      case I2L:
+      case I2F:
+      case I2D:
+      case L2I:
+      case L2F:
+      case L2D:
+      case F2I:
+      case F2L:
+      case F2D:
+      case D2I:
+      case D2L:
+      case D2F:
+      case I2B:
+      case I2C:
+      case I2S:
+        // do nothing
+        break;
+      case LCMP:
+      case FCMPL:
+      case FCMPG:
+      case DCMPL:
+      case DCMPG:
+        sv = constraintManager.createSymbolicValue(instruction);
+        pop = programState.unstackValue(2);
+        Preconditions.checkState(pop.values.size() == 2, "CMP needs 2 values on stack");
+        programState = pop.state.stackValue(sv).addConstraint(sv, ObjectConstraint.NOT_NULL);
+        break;
+      case IRETURN:
+      case LRETURN:
+      case FRETURN:
+      case DRETURN:
+      case ARETURN:
+        programState.storeExitValue();
+        programState = programState.unstackValue(1).state;
+        break;
+      case RETURN:
+        // do nothing
+        break;
+      case GETSTATIC:
+        // TODO SONARJAVA-2510 associated symbolic value with symbol
+        programState = programState.stackValue(constraintManager.createSymbolicValue(instruction));
+        break;
+      case PUTSTATIC:
+        pop = programState.unstackValue(1);
+        programState = pop.state;
+        break;
+      case GETFIELD:
+        pop = programState.unstackValue(1);
+        Preconditions.checkState(pop.values.size() == 1, "GETFIELD needs 1 value on stack");
+        programState = pop.state.stackValue(constraintManager.createSymbolicValue(instruction));
+        break;
+      case PUTFIELD:
+        pop = programState.unstackValue(2);
+        Preconditions.checkState(pop.values.size() == 2, "PUTFIELD needs 2 values on stack");
+        programState = pop.state;
+        break;
+      case INVOKEVIRTUAL:
+      case INVOKESPECIAL:
+      case INVOKESTATIC:
+      case INVOKEINTERFACE:
+        if (handleMethodInvocation(instruction)) {
+          // when yields are available, do not execute post check on this node
+          return;
         }
         break;
+      case INVOKEDYNAMIC:
+        // TODO SONARJAVA-2512
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135)